### PR TITLE
Fix langlib toStream() method issue on tuples

### LINKED ISSUE #32265: Issue with `toStream()` on tuples and finite (singleton) types
**Description:**
$title.

**Steps to reproduce:**
**Case I**
```ballerina
public function main() {
    [int, string] x = [1, "hello"];

    stream<int|string> y = x.toStream();
    stream<int|string> z = <stream<int|string>> y; // Panics - incompatible types: 'stream<()>' cannot be cast to 'stream<(int|string)>'
}
```

**Case II**
```ballerina
type Type [1, "hello"]|[2];

public function main() {
    Type x = [1, "hello"];

    stream<1|"hello"|2> y = x.toStream();
    stream<1|"hello"|2> z = <stream<1|"hello"|2>> y; // Panics - incompatible types: 'stream<()>' cannot be cast to 'stream<>'
}
```

**Affected Versions:**
slbeta2, slbeta3-SNAPSHOT

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title

Fixes #32265

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java [hunk 1]
```diff
@@ -42,16 +44,25 @@ public static BTypedesc getElementType(Object td) {
...
+        switch (type.getTag()) {
+            case TypeTags.ARRAY_TAG:
+                return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
+            case TypeTags.TUPLE_TAG:
+                return ValueCreator.createTypedescValue(
+                        TypeCreator.createUnionType(((TupleType) type).getTupleTypes()));
+            case TypeTags.STREAM_TAG:
+                return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
+            case TypeTags.TABLE_TAG:
+                return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
```
Comment (warunalakshitha) [inline langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java:56]: Lets write test for tables as well.

Comment (Nadeeshan96) [reply langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java:56]: I will look into this.

Comment (Nadeeshan96) [reply langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java:56]: It seems a ballerina code which covers the table case cannot be written. Even for the below code, this did not get covered. @pcnfernando 
```bal
table<map<string>> & readonly tbl = table [
         {a: "x", b: "3"},
         {a: "z", b: "5"}
];

function testToStreamOnImmutableArray() {
    (table<map<string>>)[] & readonly arr = [tbl];
    stream<table<map<string>> & readonly> strm = arr.toStream();

    stream<table<map<string>> & readonly> castedstrm = <stream<table<map<string>> & readonly>> strm;
}
```
I removed the table case, and ran all tests by building, and it was successful. https://github.com/Nadeeshan96/ballerina-lang/pull/6/files

Comment (pcnfernando) [reply langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java:56]: We have restricted usage of toStream() to arrays. Hence, +1 to remove the table case.

Comment (warunalakshitha) [reply langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java:56]: +1

**Comment by Nadeeshan96** [comment]
Created new PR https://github.com/ballerina-platform/ballerina-lang/pull/32469 to 2.0.0-beta3-dev branch, with the changes. Hence closing this.

**Pull request closed without merging**